supertokens_python/recipe/session/utils.py

```diff
@@ -418,4 +463,99 @@ def validate_and_normalise_user_input(
+    session: SessionContainer,
+    override_global_claim_validators: Optional[
+        Callable[
+            [List[SessionClaimValidator], SessionContainer, Dict[str, Any]],
+            MaybeAwaitable[List[SessionClaimValidator]],
+        ]
+    ],
+    user_context: Dict[str, Any],
+) -> List[SessionClaimValidator]:
+    from .recipe import SessionRecipe
+
+    claim_validators_added_by_other_recipes = (
+        SessionRecipe.get_claim_validators_added_by_other_recipes()
+    )
+    global_claim_validators = await resolve(
+        SessionRecipe.get_instance().recipe_implementation.get_global_claim_validators(
+            session.get_user_id(),
+            claim_validators_added_by_other_recipes,
+            user_context,
+        )
     )
+
+    if override_global_claim_validators is not None:
+        return await resolve(
+            override_global_claim_validators(
+                global_claim_validators, session, user_context
+            )
+        )
+
+    return global_claim_validators
+
+
+async def update_claims_in_payload_if_needed(
+    claim_validators: List[SessionClaimValidator],
+    new_access_token_payload: Dict[str, Any],
+    user_id: str,
+    user_context: Dict[str, Any],
+):
+    for validator in claim_validators:
+        log_debug_message(
+            "update_claims_in_payload_if_needed checking %s", validator.id
+        )
+        if (
+            hasattr(validator, "claim")
+            and (validator.claim is not None)
+            and (
+                await resolve(
+                    validator.should_refetch(new_access_token_payload, user_context)
+                )
+            )
+        ):
+            log_debug_message(
+                "update_claims_in_payload_if_needed refetching %s", validator.id
+            )
+            value = await resolve(validator.claim.fetch_value(user_id, user_context))
+            log_debug_message(
+                "update_claims_in_payload_if_needed %s refetch res %s",
+                validator.id,
+                json.dumps(value),
+            )
+            if value is not None:
+                new_access_token_payload = validator.claim.add_to_payload_(
+                    new_access_token_payload,
+                    value,
+                    user_context,
+                )
+
+    return new_access_token_payload
+
+
+async def validate_claims_in_payload(
+    claim_validators: List[SessionClaimValidator],
+    new_access_token_payload: Dict[str, Any],
+    user_context: Dict[str, Any],
+):
+    validation_errors: List[ClaimValidationError] = []
+    for validator in claim_validators:
+        claim_validation_res = await validator.validate(
+            new_access_token_payload, user_context
+        )
+        log_debug_message(
+            "validate_claims_in_payload %s validate res %s",
+            validator.id,
+            json.dumps(claim_validation_res),
+        )
+        if not claim_validation_res.get("isValid"):
```

Comment:
why not just !claim_validation_res.isValid

Reply:
TODO. Will convert dict into an object..

Reply:
Done